clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -263,9 +272,29 @@ def create_sessions_tsv_file(
 
 
 def _clean_exam_dates(
-    rid: str, exam_dates: List[str], visit_codes: List[str], clinical_data_dir: Path
+    rid: int,
+    exam_dates: List[Union[str, int]],
+    visit_codes: List[str],
+    clinical_data_dir: Path,
 ) -> List[str]:
-    """Clean the exam dates when necessary by trying to compute them from other sources."""
+    """Clean the exam dates when necessary by trying to compute them from other sources.
```

Comment:
This function is supposed to return a list of strings but can actually return a list with "-4" values if it fails to find the dates in the other sources.
Also, using a code value to encode missing values isn't great. It'd be better to return `List[Optional[str]]` with missing values encoded as Nones.
Even better, we could return `List[Optional[datetime.date]]` since those strings are actually dates...
WDYT ?

Reply:
I totally agree with you. This function is used in `create_sessions...` where -4 for missing values are then exchanged for "n/a"... I think I will keep as is for this PR and change when I refactor `create_sessions...` to avoid changing too many things right away. I'm keeping in mind this comment for when I do this :-)

Reply:
Alright, sounds good. 👍 
Could you still update the return type to be `List[Union[str, int]]` then ?